10_code/threshold.py

```diff
@@ -198,5 +198,199 @@
+
+#add fips to population
+
+pop_fl_final_list = []
+for state in fl_states_names:
+    tmp = fl_pop[(fl_pop["State"] == state)]
+    currstate = states_FL[state]
+    tmp_fips = fips_fl[(fips_fl["state"] == currstate)]
+    tmp_merge = pd.merge(tmp,tmp_fips, on = "County", how = "left", indicator=True)
+    pop_fl_final_list.append(tmp_merge)
+
+
+pop_fl_final = pd.concat(pop_fl_final_list)
+pop_fl_final_clean = pop_fl_final[["state", "fips", "County", "year", "population"]].copy()
+pop_fl_final_clean.sort_values(by = ["state", "fips"], inplace=True)
+
+#✅ both 
+assert (len(pop_fl_final_clean["fips"].unique())*(len(pop_fl_final_clean["year"].unique()))) == len(pop_fl_final_clean["fips"]) 
+
+#merge with complete drug deaths 
+fl_merge_all_data = pd.merge(fl_overdose_complete,pop_fl_final_clean, on = ["fips","year"], how="left", indicator=True)
+fl_merge_all_data["_merge"].value_counts()
+fl_data_to_use = fl_merge_all_data[["state_x", "fips", "year", "County" , "population" , "Deaths"]].copy()
+fl_data_to_use.rename(columns={"state_x" : "state"}, inplace=True)
+#pre data
+fl_data_pre = fl_data_to_use[(fl_data_to_use["year"] <= 2010)]
+fl_data_pre.to_csv('/Users/lorna/Documents/MIDS 2022/First Semester/720 Practicing Data Science/Final Project/final project work/pds-2022-leep/20_intermediate_files/fl_pre.csv', index=False)
+
+
+#post data 
+fl_data_post = fl_data_to_use[(fl_data_to_use["year"] >= 2011)]
+
+fl_data_post.to_csv('/Users/lorna/Documents/MIDS 2022/First Semester/720 Practicing Data Science/Final Project/final project work/pds-2022-leep/20_intermediate_files/fl_post.csv', index=False)
+
+
+
+###############################################################################################################
+""" Turn into function to do WA"""
+
+#🚩 make function
+#Florida Overdose
+WA_states = ["WA","LA","MD","OK","IN","ID","MN","NE","NV","VA"]
+WA_states_names = ["Washington", "Louisiana", "Maryland", "Oklahoma", "Indiana", "Idaho","Minnesota","Nebraska","Nevada","Virginia"]
+ 
+states_WA = {
+    "Washington": "WA",
+    "Louisiana": "LA",
+    "Maryland": "MD",
+    "Oklahoma": "OK",
+    "Indiana": "IN",
+    "Idaho": "ID",
+    "Minnesota": "MN",
+    "Nebraska": "NE",
+    "Nevada": "NV",
+    "Virginia": "VA"
+}
+
+WA_overdose = overdose_filtered[overdose_filtered["State"].isin(WA_states)].copy()
+WA_overdose.rename(columns={"County Code": "fips","State":"state"}, inplace=True)
+WA_overdose["fips"] = WA_overdose["fips"].astype(np.int64)
+WA_overdose["year"] = WA_overdose["Year"].astype(np.int64)
+WA_overdose.sort_values(by = ["state", "fips"], inplace=True)
+
+#drop years outside 2006 and 2014 range
+WA_overdose_range = WA_overdose[(WA_overdose["year"].between(2006, 2014))]
+
+#keep only those with full 9 years
+fips_in_WA = WA_overdose_range["fips"].unique()
+
+WA_compelete_list = []
+for county in fips_in_WA:
+    tmp = WA_overdose_range[(WA_overdose_range["fips"] == county)]
+    if len(tmp["year"]) == 9:
```

Comment:
Please make sure that Washington overdose dataset has 9 unique years. Since, its filtered from overdose filtered which has years from 2003 to 2015